packages/manager/src/features/IAM/Shared/RemoveAssignmentConfirmationDialog/RemoveAssignmentConfirmationDialog.tsx

```diff
@@ -0,0 +1,97 @@
+import { ActionsPanel, Notice, Typography } from '@linode/ui';
+import { useSnackbar } from 'notistack';
+import React from 'react';
+import { useParams } from 'react-router-dom';
+
+import { ConfirmationDialog } from 'src/components/ConfirmationDialog/ConfirmationDialog';
+import {
+  useAccountUserPermissions,
+  useAccountUserPermissionsMutation,
+} from 'src/queries/iam/iam';
+
+import { deleteUserEntity, type EntitiesRole } from '../utilities';
+
+interface Props {
+  onClose: () => void;
+  onSuccess?: () => void;
+  open: boolean;
+  role: EntitiesRole | undefined;
+}
+
+export const RemoveAssignmentConfirmationDialog = (props: Props) => {
+  const { onClose: _onClose, onSuccess, open, role } = props;
+  const { username } = useParams<{ username: string }>();
+
+  const { enqueueSnackbar } = useSnackbar();
+
+  const {
+    error,
+    isPending,
+    mutateAsync: updateUserPermissions,
+    reset,
+  } = useAccountUserPermissionsMutation(username);
+
+  const { data: assignedRoles } = useAccountUserPermissions(username ?? '');
+
+  const onClose = () => {
+    reset(); // resets the error state of the useMutation
+    _onClose();
+  };
+
+  const onDelete = async () => {
+    const roleName = role!.role_name;
+    const entityId = role!.entity_id;
+    const entityType = role!.entity_type;
+
+    const updatedUserEntityRoles = deleteUserEntity(
+      assignedRoles!.entity_access,
+      roleName,
+      entityId,
+      entityType
+    );
+
+    await updateUserPermissions({
+      ...assignedRoles!,
+      entity_access: updatedUserEntityRoles,
+    });
+
+    enqueueSnackbar(`Entity removed`, {
+      variant: 'success',
+    });
+    if (onSuccess) {
+      onSuccess();
+    }
+    onClose();
+  };
```

Comment:
Optional: Improve readability,  early return if role or assignedRoles is undefined.
```suggestion
const onDelete = async () => {
  if (!role || !assignedRoles) return;

  const { role_name, entity_id, entity_type } = role;

  const updatedUserEntityRoles = deleteUserEntity(
    assignedRoles.entity_access,
    role_name,
    entity_id,
    entity_type
  );

  await updateUserPermissions({
    ...assignedRoles,
    entity_access: updatedUserEntityRoles,
  });

  enqueueSnackbar(`Entity removed`, {
    variant: 'success',
  });

  onSuccess?.();
  onClose();
};

```